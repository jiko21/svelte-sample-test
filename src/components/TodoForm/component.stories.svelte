<script>
  import { Meta, Template, Story } from '@storybook/addon-svelte-csf';
  import TodoForm from './component.svelte';
</script>

<Meta
  title="components/TodoForm"
  component={TodoForm}
  argTypes={{
    add: { action: 'add' },
  }}
/>

<!-- More on component templates: https://storybook.js.org/docs/svelte/writing-stories/introduction#using-args -->
<Template let:args>
  <TodoForm {...args} on:add={args.add} />
</Template>

<!-- More on args: https://storybook.js.org/docs/svelte/writing-stories/args -->
<Story
  name="Default"
  args={{}}
/>
