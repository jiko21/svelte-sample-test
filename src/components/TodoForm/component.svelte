<script lang="ts">
  import Button from '../Button';
  import Input from '../Input';
  import { createEventDispatcher } from 'svelte';

  let input = '';
  const dispatch = createEventDispatcher();
  const handleSubmit = () => {
    if (input.length > 0) {
      dispatch('add', {
        value: input,
      });
      input = '';
    }
  };
  $: submittable = input.length > 0;
</script>

<div class="todo-form">
  <Input type="text" bind:value='{input}' testId="todo-input" />
  <Button on:click={handleSubmit} label='ADD' disabled='{!submittable}' testId="submit" />
</div>
