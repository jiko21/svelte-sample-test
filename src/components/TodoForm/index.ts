import TodoForm from './component.svelte';

export default TodoForm;
