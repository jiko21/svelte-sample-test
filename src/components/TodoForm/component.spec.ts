import { render, fireEvent, screen } from '@testing-library/svelte';
import TodoForm from './component.svelte';

const dispatchMock = jest.fn();

jest.mock('svelte', () => {
  const origin = jest.requireActual('svelte');

  return {
    ...origin,
    createEventDispatcher: () => dispatchMock,
  };
});

describe('components/Input', () => {
  it('correctly render', () => {
    const component = render(TodoForm);
    expect(component.container).toMatchSnapshot();
  });

  it('change value and active button', () => {
    const component = render(TodoForm);
    const input: any = screen.getByTestId('todo-input');
    fireEvent.change(input, {
      target: {
        value: 'test',
      },
    });
    expect(input.value).toBe('test');
    expect(component.container).toMatchSnapshot();
  });
});
