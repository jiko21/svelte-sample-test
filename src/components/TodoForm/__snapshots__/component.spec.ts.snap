// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input change value and active button 1`] = `
<body>
  <div>
    <div
      class="todo-form"
    >
      <input
        class="svelte-9hrj2z"
        data-testid="todo-input"
        type="text"
      />
       
      <button
        class="disabled svelte-1ohb8zt"
        data-testid="submit"
        disabled=""
      >
        ADD
      </button>
    </div>
  </div>
</body>
`;

exports[`components/Input correctly render 1`] = `
<body>
  <div>
    <div
      class="todo-form"
    >
      <input
        class="svelte-9hrj2z"
        data-testid="todo-input"
        type="text"
      />
       
      <button
        class="disabled svelte-1ohb8zt"
        data-testid="submit"
        disabled=""
      >
        ADD
      </button>
    </div>
  </div>
</body>
`;
